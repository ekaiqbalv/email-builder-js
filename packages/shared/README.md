# @usewaypoint/shared

Shared utilities for the email builder.
