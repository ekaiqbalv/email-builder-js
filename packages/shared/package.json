{
  "name": "@usewaypoint/shared",
  "version": "0.0.1",
  "description": "Shared utilities for the email builder",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    }
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "npx tsup src/index.ts --format cjs,esm --dts"
  },
  "author": "Eka Iqbal Virgiawan",
  "license": "MIT",
  "devDependencies": {
    "@types/jest": "^29.5.14",
    "jest": "^29.7.0",
    "ts-jest": "^29.2.5"
  }
}
