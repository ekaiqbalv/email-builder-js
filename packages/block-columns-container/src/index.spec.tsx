import React from 'react';

import { matchers } from '@emotion/jest';
import { render } from '@testing-library/react';
import { BREAKPOINTS } from '@usewaypoint/shared';

import { ColumnsContainer } from '.';

expect.extend(matchers);

describe('block-columns-container', () => {
  it('renders with default values', () => {
    expect(render(<ColumnsContainer />).asFragment()).toMatchSnapshot();
  });

  const testColumnLayout = (columnsCount: 2 | 3) => {
    const sampleColumns = [<>bread</>, <>tomato</>, <>lettuce</>];
    const paddingConfig = {
      2: {
        left: ['0px', '6px'],
        right: ['6px', '0px'],
      },
      3: {
        left: ['0px', '4px', '8px'],
        right: ['8px', '4px', '0px'],
      },
    };

    describe(`columnsCount ${columnsCount}`, () => {
      it('renders column children', () => {
        expect(
          render(<ColumnsContainer props={{ columnsCount }} columns={sampleColumns} />).asFragment()
        ).toMatchSnapshot();
      });

      it('applies correct styles', () => {
        const { container } = render(
          <ColumnsContainer
            props={{
              columnsGap: 12,
              columnsCount,
            }}
            columns={sampleColumns}
          />
        );

        const tdElements = container.querySelectorAll('td');
        const mediaQuery = `(min-width: ${BREAKPOINTS.sm}px)`;
        const config = paddingConfig[columnsCount];

        tdElements.forEach((td, index) => {
          expect(td).toHaveStyleRule('display', 'block');
          expect(td).toHaveStyleRule('display', 'table-cell!important', { media: mediaQuery });

          expect(td).toHaveStyleRule('padding-bottom', index === tdElements.length - 1 ? 'none' : '12px');

          expect(td).toHaveStyleRule('padding-bottom', '0px!important', { media: mediaQuery });
          expect(td).toHaveStyleRule('padding-left', `${config.left[index]}!important`, { media: mediaQuery });
          expect(td).toHaveStyleRule('padding-right', `${config.right[index]}!important`, { media: mediaQuery });
        });
      });
    });
  };

  testColumnLayout(2);
  testColumnLayout(3);
});
