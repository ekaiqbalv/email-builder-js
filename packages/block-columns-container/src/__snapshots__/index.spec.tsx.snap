// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block-columns-container columnsCount 2 renders column children 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            class="css-1uxtst9"
            style="box-sizing: content-box; vertical-align: middle;"
          >
            bread
          </td>
          <td
            class="css-192iasa"
            style="box-sizing: content-box; vertical-align: middle;"
          >
            tomato
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container columnsCount 3 renders column children 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            class="css-1uxtst9"
            style="box-sizing: content-box; vertical-align: middle;"
          >
            bread
          </td>
          <td
            class="css-1uxtst9"
            style="box-sizing: content-box; vertical-align: middle;"
          >
            tomato
          </td>
          <td
            class="css-192iasa"
            style="box-sizing: content-box; vertical-align: middle;"
          >
            lettuce
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container renders with default values 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            class="css-1uxtst9"
            style="box-sizing: content-box; vertical-align: middle;"
          />
          <td
            class="css-192iasa"
            style="box-sizing: content-box; vertical-align: middle;"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
